### PR TITLE
Added zipObject to Array.swift

### PR DESCRIPTION
**Pull request by Daniel-Lopez**

### CODE REVIEW

File: Sources/Array.swift [hunk 1]
```diff
@@ -213,6 +213,26 @@ public extension Array {
...
+    ///
+    /// :param values The array of values.
+    /// :return Dictionary based on the keys and values passed in order.
+    public func zipObject<T>(values: [T]) -> [Element:T] {
+        
+        var dict = [Element:T]()
+        
+        guard self.count == values.count else { return dict }
+        
+        for (i,key) in self.enumerate() {
```
Comment (ankurp) [inline Sources/Array.swift]: Can you add a space after the comma `(i, key)`

**Comment by ankurp** [comment]
Thanks for the Pull Request. Left one small comment.

File: Sources/Array.swift [hunk 2]
```diff
@@ -213,6 +213,26 @@ public extension Array {
   
 }
 
+extension Array where Element: Hashable {
+  
+    /// Creates an object composed from arrays of keys and values.
+    ///
+    /// :param values The array of values.
+    /// :return Dictionary based on the keys and values passed in order.
+    public func zipObject<T>(values: [T]) -> [Element:T] {
```
Comment (ankurp) [inline Sources/Array.swift:222]: Can you use the Dollar zipObject method instead https://github.com/ankurp/Dollar/blob/master/Sources/Dollar.swift#L1308

``` swift
public func zipObject<T>(values: [T]) -> [Element:T] {
    return $.zipObject(self, values)
}
```

**Comment by ankurp** [comment]
Also I am thinking of fixing the implementation of Dollar zipObject by adding the guard clause.

**Comment by Daniel-Lopez** [comment]
Yes, that makes good sense. Done.

**Comment by ankurp** [comment]
Thanks merged.

**Pull request merged**